src/opendlv-video-h264-encoder.cpp

```diff
@@ -83,31 +137,63 @@ int32_t main(int32_t argc, char **argv) {
                 parameters.iPicHeight = HEIGHT;
                 parameters.uiIntraPeriod = GOP;
                 parameters.iTargetBitrate = BITRATE;
-                parameters.iMaxBitrate = BITRATE_MAX;
-                parameters.iRCMode = RC_MODES::RC_QUALITY_MODE;
                 parameters.iSpatialLayerNum = 1;
                 parameters.iTemporalLayerNum = 1;
-                parameters.iLoopFilterDisableIdc = 0;
                 parameters.iLtrMarkPeriod = 30;
-                parameters.iMultipleThreadIdc = 1; // 1 = disable multi threads.
-                parameters.iEntropyCodingModeFlag = 0; // 0 = CAVLC, 1 = CABAC (not supported in BaseLine profile).
-                parameters.iComplexityMode = ECOMPLEXITY_MODE::LOW_COMPLEXITY;
-                parameters.bEnableAdaptiveQuant = 1;
-                parameters.bEnableBackgroundDetection = 1;
-                parameters.bEnableDenoise = 1;
-                parameters.bEnableFrameSkip = 0;
-                parameters.bEnableLongTermReference = 0;
-                parameters.bEnableSceneChangeDetect = 1;
-                parameters.bPrefixNalAddingCtrl = 0; // do nod add NAL prefixes.
-                parameters.eSpsPpsIdStrategy = EParameterSetStrategy::CONSTANT_ID;
+                parameters.iMultipleThreadIdc = I_MULTIPLE_THREADS; // 1 = disable multi threads.
 
                 parameters.sSpatialLayers[0].iVideoWidth = parameters.iPicWidth;
                 parameters.sSpatialLayers[0].iVideoHeight = parameters.iPicHeight;
                 parameters.sSpatialLayers[0].fFrameRate = parameters.fMaxFrameRate;
                 parameters.sSpatialLayers[0].iSpatialBitrate = parameters.iTargetBitrate;
-                parameters.sSpatialLayers[0].iMaxSpatialBitrate = parameters.iMaxBitrate;
+                parameters.sSpatialLayers[0].iMaxSpatialBitrate = I_BITRATE_MAX;
                 parameters.sSpatialLayers[0].sSliceArgument.uiSliceMode = SliceModeEnum::SM_SIZELIMITED_SLICE;
                 parameters.sSpatialLayers[0].sSliceArgument.uiSliceNum = 1;
+
+                /*
+                 * Thesis parameters
+                 * https://github.com/cisco/openh264/wiki/TypesAndStructures
+                 * https://github.com/cisco/openh264/blob/master/codec/encoder/core/inc/param_svc.h#L132
+                 */
+                if (I_NUM_REF_FRAME==0){parameters.iNumRefFrame = AUTO_REF_PIC_COUNT;}
```

Comment:
Please add a blank and a new line around the `{` brackets for readability.

Reply:
Fixed in 7a978cd